File: src/BooKing.Web/src/app/dtos/reservation.dto.ts
import { ApartmentDto } from "./apartment.dto";
import { ReservationEventsDto } from "./reservationevents.dto";

export class ReservationDto {
  id: string;
  userId: string;
  apartmentId: string;
  apartment: ApartmentDto;
  duration: DateRange;
  priceForPeriod: number;
  cleaningFee: number;
  totalPrice: number;
  statusDescription: string;
  statusValue: number;
  createdOnUtc: Date;
  confirmedOnUtc?: Date;
  rejectedOnUtc?: Date;
  completedOnUtc?: Date;
  cancelledOnUtc?: Date;
  events: ReservationEventsDto[];

  constructor(
    id: string,
    userId: string,
    apartmentId: string,
    apartment: ApartmentDto,
    duration: DateRange,
    priceForPeriod: number,
    cleaningFee: number,
    totalPrice: number,
    statusDescription: string,
    statusValue: number,
    createdOnUtc: Date,
    events: ReservationEventsDto[],
    confirmedOnUtc?: Date,
    rejectedOnUtc?: Date,
    completedOnUtc?: Date,
    cancelledOnUtc?: Date,
  ) {
    this.id = id;
    this.userId = userId;
    this.apartmentId = apartmentId;
    this.apartment = apartment;
    this.duration = duration;
    this.priceForPeriod = priceForPeriod;
    this.cleaningFee = cleaningFee;
    this.totalPrice = totalPrice;
    this.statusDescription = statusDescription;
    this.statusValue = statusValue;
    this.createdOnUtc = createdOnUtc;
    this.confirmedOnUtc = confirmedOnUtc;
    this.rejectedOnUtc = rejectedOnUtc;
    this.completedOnUtc = completedOnUtc;
    this.cancelledOnUtc = cancelledOnUtc;
    this.events = events;
  }
}

export class DateRange {
  start: Date;
  end: Date;
  lengthInDays: number;

  constructor(
    start: Date,
    end: Date,
    lengthInDays: number
  ){
    this.start = start;
    this.end = end;
    this.lengthInDays = lengthInDays;
  }
}

File: src/BooKing.Reserve/BooKing.Reserve.Application/Dtos/ReservationDto.cs
using BooKing.Reserve.Domain.ValueObjects;

namespace BooKing.Reserve.Application.Dtos;
public class ReservationDto
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid ApartmentId { get; set; }
    public GetApartmentDto Apartment { get; set; }
    public DateRange Duration { get; set; }
    public decimal PriceForPeriod { get; set; }
    public decimal CleaningFee { get; set; }
    public decimal TotalPrice { get; set; }
    public string StatusDescription { get; set; }
    public int StatusValue { get; set; }
    public DateTime CreatedOnUtc { get; set; }
    public DateTime? ConfirmedOnUtc { get; set; }
    public DateTime? RejectedOnUtc { get; set; }
    public DateTime? CompletedOnUtc { get; set; }
    public DateTime? CancelledOnUtc { get; set; }
}
